feature-libs/cart/base/components/abstract-order-context/abstract-order-context.model.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { AbstractOrderType } from '@spartacus/cart/base/root';
+
+import { Observable } from 'rxjs';
+
+/**
+ * Context for transporting information about an abstract order (active cart, saved cart, order, quote).
+ */
+@Injectable()
+export abstract class AbstractOrderContext {
+  readonly type$: Observable<AbstractOrderType>;
+  readonly id$: Observable<string | undefined>;
```

Comment:
@ChristophHi

Reply:
Id is only required for savedCarts, orders and quotes. For the active cart, an id is not needed as from the component layer we can address the active cart directly without specifying its id.
That also allows to leave all the views untouched that use the cartItemList and deal with the active cart. We just omit the context here. For child views using the context it is optional; in case it's not provided from outside, we default to type CART (and obviously here cannot, and don't need to, provide an id)

Reply:
I will also mention this in the JSDoc for the `AbstractOrderKey`.
Do you see an issue with that @Platonn ?

Reply:
Thank you for your explanation. It's very good. Yes, let's leave it also in JSDoc of `AbstractOrderKey.id`. Do you think that it would be beneficial to leave a comment also in JSDoc of `AbstractOrderContext.id`?

Reply:
Yes, agree